spyder/plugins/ipythonconsole/widgets/shell.py

```diff
@@ -311,6 +311,74 @@ def request_env(self):
         self.call_kernel(
             interrupt=True, callback=self.sig_show_env.emit).get_env()
 
+    def set_paging(self, use_pager):
+        """Set pager option."""
+        self.paging = 'inside' if use_pager else 'none'
+
+    def set_show_calltips(self, show_calltips):
+        """Enable/Disable showing calltips."""
+        self.enable_calltips = show_calltips
+
+    def set_buffer_size(self, buffer_size):
+        """Set buffer size for the shell."""
+        self.buffer_size = buffer_size
+
+    def set_completion_type(self, completion_type):
+        """Set completion type (Graphical, Terminal, Plain) for the shell."""
+        self.gui_completion = completion_type
+
+    def set_in_prompt(self, in_prompt):
+        """Set appereance of the In prompt."""
+        self.in_prompt = in_prompt
+
+    def set_out_prompt(self, out_prompt):
+        """Set appereance of the Out prompt."""
+        self.out_prompt = out_prompt
+
+    def get_matplotlib_backend(self):
+        """Call kernel to get current backend."""
+        return self.call_kernel(
+            interrupt=True,
+            blocking=True).get_matplotlib_backend()
+
+    def set_matplotlib_backend(self, backend_option, pylab=False):
+        """Set matplotlib backend given a backend name."""
+        self.call_kernel(
+            interrupt=True).set_matplotlib_backend(
```

Comment:
I don't think that will work: When you call the kernel with `interrupt=True`, it will run the code in a thread. This function need to be run in the main thread. In https://github.com/spyder-ide/spyder/pull/14199, I solved it by:
```
backend = "inline" if self._gui != "inline" else "auto"
self.current_shellwidget().execute("%matplotlib " + backend)
```
Execute will be executed whenever a prompt becomes available.

Reply:
Also avilable with hidden=True

Reply:
Sorry, I missed this one @impact27. I'll follow your suggestion in another PR.